src/realm/query_expression.cpp

```diff
@@ -145,31 +134,25 @@ void LinkMap::map_links(size_t column, size_t row, LinkMapFunction lm) const
                 values.init_from_parent();
 
                 // Iterate through values and insert all link values
-                values.for_all([&](Mixed val) {
-                    if (val.is_type(type_TypedLink)) {
-                        auto link = val.get_link();
-                        REALM_ASSERT(link.get_table_key() == this->m_tables[column + 1]->get_key());
-                        auto k = link.get_obj_key();
-                        if (!k.is_unresolved()) {
-                            if (last)
-                                lm(k);
-                            else
-                                map_links(column + 1, k, lm);
+                values.traverse([&](BPlusTreeNode* node, size_t) {
```

Comment:
You should preserve `for_all` here

Reply:
for_all() does not support exiting early, while LinkMap is supposed to (this was a pre-existing bug that it sometimes did not).

Reply:
I've added support for early exits from for_all() and switched this back to using it.